lib/resolver.js

```diff
@@ -177,7 +193,7 @@ class SchemaResolver {
     resolveOneOf(schema) {
         Hoek.assert(Util.isArray(schema.oneOf), 'Expected oneOf to be an array.');
 
-        return this.joi.alternatives(schema.oneOf.map(schema => this.resolve(schema))).match('any');
+        return this.joi.alternatives(schema.oneOf.map(schema => this.resolve(schema))).match('one');
```

Comment:
Is this a breaking change?

Reply:
yes. this produces different results than the old incorrect `any` behavior.